### PR TITLE
Adding documentation on configuring different endpoint types from the API Controller 3.1.0

### PR DESCRIPTION
**Pull request by BLasan**
## Purpose
Documentation on setting up `api_params` with multi endpoint configurations.

## Goals

- Add documentation for configuring the four (4) main types of endpoint types using the api_params.yaml file from the API Controller.
- Update `Configuring Environment Specific Parameters` document with added configuration parameters.

## User stories
Setting up the endpoints with respect to the below nine (9) user scenarios have been documented for the reference of the user.

1. HTTP/REST Endpoint - Without load balancing or failover (the usual way)
2. HTTP/REST Endpoint - With load balancing
3. HTTP/REST Endpoint - With failover
4. HTTP/SOAP Endpoint - Without load balancing or failover
5. HTTP/SOAP Endpoint - With load balancing
6. HTTP/REST Endpoint - With failover
7. Dynamic Endpoint
8. AWS Lambda - Using IAM role-supplied temporary AWS credentials
9. AWS Lambda - Using stored AWS credentials

## Related PRs
- https://github.com/wso2/docs-apim/pull/1391

## Related Issues
 - https://github.com/wso2/api-manager/issues/172

### CODE REVIEW

File: en/mkdocs.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ theme:
     tabs: true
   language: 'en'
 #Breaks build if there's a warning
-strict: true
+strict: false
```
[the remark this answers is not in the document]
Comment (BLasan) [reply en/mkdocs.yml]: Fixed by [fcf676c](https://github.com/wso2/docs-apim/pull/5955/commits/fcf676ca0073380d674a4afd74c8e5b69fecdb1b)

**Comment by BLasan** [comment]
@wasuradananjith updated the `configuring-environment-specific-parameters` doc

File: en/docs/learn/api-controller/advanced-topics/configuring-environment-specific-parameters.md [hunk 2]
```diff
@@ -59,7 +87,19 @@ The following code snippet contains sample configuration of the parameter file.
               production:
                   url: 'https://test.wso2.com'
                   config:
-                      retryTimeOut: $RETRY
+                    factor: 3
```
Comment (wasuradananjith) [inline en/docs/learn/api-controller/advanced-topics/configuring-environment-specific-parameters.md:90]:
```suggestion
                      factor: 3
```
Let's indent all the fields like this? (the earlier _retryTimeOut: $RETRY_ had correct indentation I guess)

File: en/docs/learn/api-controller/advanced-topics/configuring-environment-specific-parameters.md [hunk 3]
```diff
@@ -59,7 +87,19 @@ The following code snippet contains sample configuration of the parameter file.
...
+                    suspendErrorCode: 
+                        - "101504"
+                        - "101501"
+                    retryTimeOut: $RETRY
+                    retryDelay: 23000
+                    retryErroCode:
+                        - "101503" 
+                        - "101504"
+                    actionSelect: discard
+                    actionDuration: 75000
```
Comment (wasuradananjith) [inline en/docs/learn/api-controller/advanced-topics/configuring-environment-specific-parameters.md:102]:
```suggestion
                      factor: 3
                      suspendMaxDuration: 25000
                      suspendDuration: 45000
                      suspendErrorCode: 
                          - "101504"
                          - "101501"
                      retryTimeOut: $RETRY
                      retryDelay: 23000
                      retryErroCode:
                          - "101503" 
                          - "101504"
                      actionSelect: discard
                      actionDuration: 75000
```
```suggestion
                      factor: 3
```
Let's indent all the fields like this? (the earlier _retryTimeOut: $RETRY_ had correct indentation I guess)

Comment (BLasan) [reply en/docs/learn/api-controller/advanced-topics/configuring-environment-specific-parameters.md]: Fixed by [dca22dd](https://github.com/wso2/docs-apim/pull/5955/commits/dca22dd60544d2d61fe44191e9c167bcb16cafed)

**Review by wasuradananjith — APPROVED** [review]

**Comment by wasuradananjith** [comment]
> @wasuradananjith updated the `configuring-environment-specific-parameters` doc

Approved. 

> @wasuradananjith updated the `configuring-environment-specific-parameters` doc

Approved. Let's merge this after the apictl 3.1.5 release is done and the artifacts are updated on the website.

**Pull request merged**